### PR TITLE
Feature signup login frontend

### PR DESCRIPTION
**Pull request by c-s-ale**
### What this PR does (required):
Adds styling for the Login/Signup pages as per the 'Sign up' mockup we received. 

### Screenshots / Videos (required):
image
Login page, I've cropped the remaining right hand side of the screen, as it is currently empty - the same is true of the screenshot of the Signup page. Let me know if I should've instead included the entire webpage!

image
Signup page

loginsignup

### Any information needed to test this feature (required):
N/A

### Any issues with the current functionality (optional):
Not exactly a problem, but: The mockup indicated that the login/signup page should be in a card centered in the middle of the screen. However, our login/signup pages will eventually be cast into the left container on the homepage

image

So, I've let the card fill the entire left-side container and style relative to the card (I believe I did that, anyway) 

I also did change /client/themes/theme.ts, which may impact others' styling if they were referencing the 'primary' variable at all. However, it was discussed in Slack - and I believe this is better suited to our primary colour since it's the primary colour in the mockups!

Thanks for bearing with me, this is my first 'official' PR on git! :D

### CODE REVIEW

File: client/src/context/useAuthContext.tsx [hunk 1]
```diff
@@ -39,7 +39,13 @@ export const AuthProvider: FunctionComponent = ({ children }): JSX.Element => {
       })
       .catch((error) => console.error(error));
   }, [history]);
+  /*
+  This code will automatically cause a redirect to the login page if
+  I attempt to navigate to the "/signup" route
 
+  Commenting out this functionality until it is implemented:
+  
+  '''
```
Comment (moffatethan) [inline client/src/context/useAuthContext.tsx:48]: We don't want to push up commented code. This sign up functionality actually still works, the issue lies with the `<Link />` tag. We are using Material-UIs link tag, this is good for anchor tags however, anchor tags refresh the page which triggers the auth context to redo it's logic causing the issue. 

To fix this, on the login page where it says "Register" make sure you utilize react-router-dom's Link component.

File: client/src/pages/Login/Login.tsx [hunk 2]
```diff
@@ -41,18 +42,26 @@ export default function Login(): JSX.Element {
       <CssBaseline />
       <Grid item xs={12} sm={8} md={7} elevation={6} component={Paper} square>
         <Box className={classes.authWrapper}>
-          <AuthHeader linkTo="/signup" asideText="Don't have an account?" btnText="Create account" />
+          {/* --Commented out until thew new authHeader is completed
+          <AuthHeader linkTo="/signup" asideText="Don't have an account?" btnText="Create account" />*/}
```
Comment (moffatethan) [inline client/src/pages/Login/Login.tsx:46]: If this is related to the routing issue, same applies here. Just make sure the `AuthHeader` is using the react-router-dom Link component.

File: client/src/pages/SignUp/SignUp.tsx [hunk 3]
```diff
@@ -42,18 +43,26 @@ export default function Register(): JSX.Element {
       <CssBaseline />
       <Grid item xs={12} sm={8} md={7} elevation={6} component={Paper} square>
         <Box className={classes.authWrapper}>
-          <AuthHeader linkTo="/login" asideText="Already have an account?" btnText="Login" />
+          {/* --Commented out until thew new authHeader is completed
+          <AuthHeader linkTo="/signup" asideText="Don't have an account?" btnText="Create account" />*/}
```
Comment (moffatethan) [inline client/src/pages/SignUp/SignUp.tsx:47]: Same here with regards to react-router-dom `Link` component

**Review by moffatethan — CHANGES_REQUESTED** [review]
Hey Chris! This is looking good. Left you some comments to address as for the layout, it's coming together.

File: client/src/pages/Login/Login.tsx [hunk 4]
```diff
@@ -41,18 +42,26 @@ export default function Login(): JSX.Element {
...
                 <Typography className={classes.welcome} component="h1" variant="h5">
-                  Welcome back!
+                  Log in
                 </Typography>
               </Grid>
             </Grid>
             <LoginForm handleSubmit={handleSubmit} />
           </Box>
-          <Box p={1} alignSelf="center" />
+          <Box p={1} alignSelf="center" padding="0" />
```
Comment (bonnieli) [inline client/src/pages/Login/Login.tsx:57]: what is this Box doing, is it just for added space? if so - i'd remove this and utilizing styling from sibling components to create this space

File: client/src/pages/SignUp/SignUpForm/useStyles.ts [hunk 5]
```diff
@@ -6,28 +6,28 @@ const useStyles = makeStyles((theme) => ({
...
-    margin: theme.spacing(3, 2, 2),
+    margin: theme.spacing(3, 2, -1),
     padding: 10,
     width: 160,
     height: 56,
     borderRadius: theme.shape.borderRadius,
     marginTop: 49,
     fontSize: 16,
-    backgroundColor: '#3a8dff',
+    backgroundColor: '#primary',
```
Comment (bonnieli) [inline client/src/pages/SignUp/SignUpForm/useStyles.ts:30]: what is #primary color here? is this a valid color? did you mean to use the theme's primary colour?

File: client/src/pages/Login/LoginForm/useStyles.ts [hunk 6]
```diff
@@ -1,33 +1,34 @@
...
-    padding: 10,
+    margin: theme.spacing(3, 2, -7),
+    padding: 0,
     width: 160,
     height: 56,
     borderRadius: theme.shape.borderRadius,
     marginTop: 49,
     fontSize: 16,
-    backgroundColor: '#3a8dff',
+    backgroundColor: 'primary',
```
Comment (bonnieli) [inline client/src/pages/Login/LoginForm/useStyles.ts:31]: is this actually changing to the string primary? what is primary colour?

File: client/src/pages/Login/Login.tsx (hunk 4 above)
Comment (c-s-ale) [reply client/src/pages/Login/Login.tsx:57]: Hello!

The box was there from the original code, I have removed it from Login.tsx, as well as Signup.tsx!

File: client/src/pages/Login/LoginForm/useStyles.ts (hunk 6 above)
Comment (c-s-ale) [reply client/src/pages/Login/LoginForm/useStyles.ts]: As with :

what is #primary color here? is this a valid color? did you mean to use the theme's primary colour?

_Originally posted by @bonnieli in https://github.com/hatchways/team-corgi/pull/42#discussion_r652087127_

I have fixed both of the references to our primary colour:

image

File: client/src/pages/SignUp/SignUp.tsx (hunk 3 above)
Comment (c-s-ale) [reply client/src/pages/SignUp/SignUp.tsx]: I have un-commented all code, as requested!

File: client/src/context/useAuthContext.tsx (hunk 1 above)
Comment (c-s-ale) [reply client/src/context/useAuthContext.tsx]: Thank you!

I have switched to the 'react-router-dom' Link and it works like a charm. That information is invaluable - thank you!

**Comment by c-s-ale** [comment]
I believe I have made the requested changes, let me know if anything else needs to be done - or if I missed anything!

**Comment by c-s-ale** [comment]
I have now completed the landing page, I believe.


https://user-images.githubusercontent.com/19699016/122248878-6469be80-ce96-11eb-9f63-c31b104a4483.mp4

Please let me know what I can/should improve!

**Comment by c-s-ale** [comment]
Realized I should've made the landing page bar a component, and switching it to that!

File: .gitignore [hunk 7]
```diff
@@ -14,6 +14,8 @@ node_modules/
 .env.development.local
 .env.test.local
 .env.production.local
+/development
+
```
Comment (moffatethan) [inline .gitignore:18]: What's this directory?

Comment (c-s-ale) [reply .gitignore:18]: This is just where I'm placing things like screenshots/screencaps to post here to git, I can make it public if you want!

Comment (moffatethan) [reply .gitignore:18]: No need, just was curious what the folder was. Fine with me 👍🏻

File: client/src/components/LandingPageBar/useStyles.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
+import { makeStyles } from '@material-ui/core/styles';
+import { theme } from '../../themes/theme';
+
+const useStyles = makeStyles(() => ({
+  sitterLoginSignup: {
+    position: 'fixed',
```
Comment (moffatethan) [inline client/src/components/LandingPageBar/useStyles.ts:6]: MUI will actually pass the `theme` as an argument, I would utilize it there as opposed to importing the theme.

Example: `makeStyles((theme: Theme) => ({ . . .}));`

File: client/src/components/Logo/useStyles.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
+import { makeStyles } from '@material-ui/core/styles';
+import { theme } from '../../themes/theme';
+
+const useStyles = makeStyles(() => ({
+  logoBox: {
```
Comment (moffatethan) [inline client/src/components/Logo/useStyles.ts:5]: As mentioned above, MUI will pass the theme as an argument.

File: client/src/components/Logo/Logo.tsx [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+import PetsIcon from '@material-ui/icons/Pets';
+
+const Logo = (): JSX.Element => {
+  const classes = useStyles();
+
+  return (
+    <Box className={classes.logoBox}>
+      <PetsIcon className={classes.logoIcon} />
+      <Typography className={classes.logoText}>LovingSitter.</Typography>
+    </Box>
```
Comment (moffatethan) [inline client/src/components/Logo/Logo.tsx:13]: We might not even need to do a Box and styles here. Refer to [Grid](https://material-ui.com/components/grid/)

File: client/src/pages/Login/Login.tsx [hunk 11]
```diff
@@ -36,25 +42,45 @@ export default function Login(): JSX.Element {
...
                 </Typography>
               </Grid>
             </Grid>
             <LoginForm handleSubmit={handleSubmit} />
           </Box>
-          <Box p={1} alignSelf="center" />
+          <Box alignSelf="center">
+            <Typography variant="button">
+              Not a member yet?
+              <span> </span>
```
Comment (moffatethan) [inline client/src/pages/Login/Login.tsx:73]: Is this here to separate the text and the link? Could this be refactor to be a bit more cleaner?

File: client/src/pages/Login/LoginForm/useStyles.ts [hunk 12]
```diff
@@ -1,33 +1,34 @@
 import { makeStyles } from '@material-ui/core/styles';
+import { SignalWifi1BarLockSharp } from '@material-ui/icons';
```
Comment (moffatethan) [inline client/src/pages/Login/LoginForm/useStyles.ts:2]: This isn't being used here

File: client/src/pages/SignUp/SignUp.tsx [hunk 13]
```diff
@@ -37,25 +42,45 @@ export default function Register(): JSX.Element {
     });
   };
 
+  //---CODE FOR ADDING A BUTTON FOR A DEMO USER: DEVELOPMENT ONLY
```
Comment (moffatethan) [inline client/src/pages/SignUp/SignUp.tsx:45]: As mentioned before, this comment isn't necessary.

File: client/src/pages/SignUp/useStyles.ts [hunk 14]
```diff
@@ -1,4 +1,5 @@
 import { makeStyles } from '@material-ui/core/styles';
+import { theme } from '../../themes/theme';
 
 const useStyles = makeStyles(() => ({
   root: {
```
Comment (moffatethan) [inline client/src/pages/SignUp/useStyles.ts:5]: Theme is passed as argument here as well

**Review by moffatethan — APPROVED** [review]
I'm going to approve this and left you some cleanup comments to address. After that this should be good to go. Good work on this!

File: client/src/components/Logo/Logo.tsx (hunk 10 above)
Comment (c-s-ale) [reply client/src/components/Logo/Logo.tsx:13]: I have swapped this to Grid, and refactored some of the other styling to make it fit

File: client/src/pages/Login/Login.tsx [hunk 15]
```diff
@@ -36,25 +42,45 @@ export default function Login(): JSX.Element {
     });
   };
 
+  //---CODE FOR ADDING A BUTTON FOR A DEMO USER: DEVELOPMENT ONLY
```
[the remark this answers is not in the document]
Comment (c-s-ale) [reply client/src/pages/Login/Login.tsx:45]: I have removed the comment!

File: client/src/pages/Login/Login.tsx (hunk 11 above)
Comment (c-s-ale) [reply client/src/pages/Login/Login.tsx:73]: I've modified this to use inline CSS to add a margin to the text

**Pull request merged**